utils.go

```diff
@@ -35,40 +35,42 @@ func logErrIfNotNil(err error) {
 	log.Error(err)
 }
 
-func toDnstap(ctx context.Context, host, protocol string, state *request.Request, reply *dns.Msg, start time.Time) error {
-	tapper := dnstap.TapperFromContext(ctx)
-	if tapper == nil {
-		return nil
-	}
+func toDnstap(f *Fanout, host string, state *request.Request, reply *dns.Msg, start time.Time) {
```

Comment:
This looks like sync up with https://github.com/coredns/coredns/blob/master/plugin/forward/dnstap.go#L16-L58

But it should be fine for now.

@miekg Let us know if we can make some stuff from the forward plugin reusable.